### PR TITLE
Add client heartbeats

### LINKED ISSUE #1655: Dask distributed client doesn't keep the connection alive
We connect client machine to a distributed dask scheduler, submit a long running task graph and eventually call progress() to wait for completion.
In this case dask client doesn't seem to keep the connection alive.
If the connection goes through ELB or any other element that has an idle timeout, the connection is disconnected and the job is dropped (in case of ELB max timeout is 4000s).

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #1655

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @pitrou for review

File: distributed/config.yaml [hunk 1]
```diff
@@ -58,3 +58,5 @@ worker-memory-terminate: 0.95  # fraction at which we terminate the worker
 
 log-length: 10000  # default length of logs to keep in memory
 log-format: '%(name)s - %(levelname)s - %(message)s'
+
+client-heartbeat-interval: 5000  # milliseconds between client heartbeats
```
Comment (pitrou) [inline distributed/config.yaml]: I'd rather have seconds everywhere :-)

File: distributed/client.py [hunk 2]
```diff
@@ -456,6 +456,8 @@ class resembles executors in ``concurrent.futures`` but also allows
     name: string (optional)
         Gives the client a name that will be included in logs generated on
         the scheduler for matters relating to this client
+    heartbeat_interval: int
+        Time in milliseconds between heartbeats to scheduler
```
Comment (pitrou) [inline distributed/client.py]: Ditto here.

File: distributed/config.yaml (hunk 1 above)
Comment (mrocklin) [reply distributed/config.yaml]: Yeah, I was thinking about this as I was writing this.  Historically we have preferred milliseconds when we're just passing through to Tornado or Bokeh and they expect seconds.  However we also use seconds when we do our own thing.  This has plenty of opportunity for confusion.

As a solution I suggest that we make a `parse_time` function and use strings like `"5000 ms"` in configuration files in the future.  

Smoothly transitioning will be tricky no matter what solution we choose.

Comment (mrocklin) [reply distributed/config.yaml]: > Tornado or Bokeh and they expect seconds

Should be 

Tornado or Bokeh and they expect milliseconds

Comment (pitrou) [reply distributed/config.yaml]: I suspect Bokeh copied Tornado's initial API mistake. But, well, no need to accumulate our own debt here :-)

Comment (mrocklin) [reply distributed/config.yaml]: Yeah, I'm in favor of changing this.  This is a larger task though and will need to be done consistently across the codebase.

Comment (mrocklin) [reply distributed/config.yaml]: @pitrou any objection to holding off on this change until we change it globally?  For the moment this choice matches current conventions in the codebase.

I'm happy to commit to changing to seconds or strings everywhere prior to the next 0.X.0 release.

**Comment by pitrou** [comment]
Ok from me.

**Comment by mrocklin** [comment]
Merging.  Thanks for the review @pitrou 

Follow up issue on time units in https://github.com/dask/distributed/issues/1660

**Pull request merged**